### DISCUSSION TITLE
Segment of road not considered for routing - why?

### DISCUSSION THREAD

**Discussion by tumik** [Project-OSRM/osrm-backend#6835] [Q&A]
# Issue

I'm seeing a routing issue in Tampere, Finland where a segment of road just seems to be missing from the database of OSRM.

You can see and test the issue easily here:
https://map.project-osrm.org/?z=17&center=61.498896%2C23.777643&loc=61.500055%2C23.774870&loc=61.500739%2C23.774629&hl=en&alt=0&srv=0
There is a segment of road missing straight between the starting and ending point, so the router finds a detour far away.
I don't see any trouble why it shouldn't be considered for routing in the OSM data.
https://www.openstreetmap.org/way/1126408079
https://www.openstreetmap.org/way/1126408080

Looking at the debug tiles the segment just seems to be missing completely:
https://map.project-osrm.org/debug/#16.71/61.49999/23.77709

![image](https://github.com/Project-OSRM/osrm-backend/assets/418911/5afef088-1233-46ee-a0a3-c63c482cf966)
![image](https://github.com/Project-OSRM/osrm-backend/assets/418911/10be8e0c-f27e-4b8e-9334-7afbf7be4873)

I don't think this is related to the demo site - the same issue occurs with my self-build OSRM with a custom profile and up-to-date map data.

# Steps to reproduce

https://map.project-osrm.org/?z=17&center=61.498896%2C23.777643&loc=61.500055%2C23.774870&loc=61.500739%2C23.774629&hl=en&alt=0&srv=0

**Comment by jcoupey** [comment]
There is a dedicated lua script useful for debugging that will help understand how a profile "sees" a given OSM way. For example one of the two you shared seems now perfectly fine:

```bash
lua debug_way.lua car 1126408079
{ 
  highway = 'primary',
  maxspeed = '40',
  name = 'Ratapihankatu',
  ['name:fi'] = 'Ratapihankatu',
  oneway = 'yes',
  surface = 'asphalt' 
}
=>
{ 
  backward_classes = {},
  backward_mode = 0,
  backward_rate = 8.8888888888889,
  backward_speed = 32,
  destinations = '',
  duration = 0,
  forward_classes = {},
  forward_mode = 1,
  forward_rate = 8.8888888888889,
  forward_speed = 32,
  is_left_hand_driving = false,
  is_startpoint = true,
  name = 'Ratapihankatu',
  road_classification = { 
    may_be_ignored = false,
    road_priority_class = 4 
  } 
}
```

Note that this way has been changed right after you posted your message: the `proposed: primary` that was removed is likely the reason why it was not included in the first place, see [this part of the profile](https://github.com/Project-OSRM/osrm-backend/blob/8526cc7d45adce9c352897acfd0366211e4f1d5e/profiles/car.lua#L130-L139).

That is really not a bug, rather a data issue. Can you re-build using a new extract including the latest changes and confirm that those ways are now included?

**Comment by tumik** [comment]
Thank you, that explains it all. The lua script seems useful.

Seems like the proposed-tag was misused on that way, causing map renderers to show it as a proper functioning way but OSRM profiles considering it as a planned but not ready for use road.

The proposed tag should always be used combined with "highway: proposed" (or amenity/railway/another main tag: proposed). Wouldn't it be better for the profile to ignore the proposed-tag and instead avoid highways with tag "highway: proposed" ? I believe that could be more consistent with the renderers and avoid cases where proposed-tag is forgotten on a way when the road is marked as finished.

**Reply by scmc24** [reply]
please i need help on how to configure lua profiles to take into account my custom road data

**Reply by tombay** [reply]
FIrstly -
https://map.project-osrm.org/?z=17&center=61.498896%2C23.777643&loc=61.500055%2C23.774870&loc=61.500739%2C23.774629&hl=en&alt=0&srv=0 
show the way correctly now..
What do you mean by custom road data? 
If you mean you have extra ".osm" data you can use osmconvert or "osmium merge" to merge your data with the original data. osmstats (http://osmstats.neis-one.org/?item=elements) gives the latest highest numbers for nodes,ways,relations so use numbers way beyond those to avoid clashes.
If you mean you have added tags to your own dataset then you probably need to add lua code. In car.lua (for example) process_way() and process_node() outline the steps taken.
